### PR TITLE
ROX-11320: Adding create policy category modal

### PR DESCRIPTION
**Pull request by alwayshooin**
## Description

Adding the policy create category modal to the policy categories page. 
Should look like the below and disable interacting with default categories. 
When creating a new category, if successful, should auto select the newly created category.

image
image
image

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: ui/apps/platform/src/Containers/PolicyCategories/PolicyCategoriesList.tsx [hunk 1]
```diff
@@ -19,15 +19,15 @@ function PolicyCategoriesList({
...
             {policyCategories.map((category) => {
-                const { id, name } = category;
+                const { id, name, isDefault } = category;
                 return (
                     <SimpleListItem
                         key={id}
                         onClick={() => {
                             setSelectedCategory(category);
                         }}
                         isActive={false}
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/PolicyCategories/PolicyCategoriesList.tsx:29]: Will `isActive` prop have a conditional value in a future pull request?

**Comment by pedrottimark** [comment]
Off the topic of this pull request, while testing `isDisabledProp` of **Create policy** button:

1. Result of **Save** existing policy category with new name does not update the list. Because backend can adjust the user name according to its naming convention, the update needs to use the response instead of the form values, correct? Maybe also update the form values?
 
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/PolicyCategories/PolicyCategorySidePanel.tsx#L33-L36

    ```js
            renamePolicyCategory(id, name)
                .then((response) => {
                    setSelectedCategory(response);
                    // TODO also update values?
                })
    ```

2. Result of **Create** new policy category needs to use response:
    * which has `id` prop to correctly support a subsequent update?
    * also because of possible backend user name adjustment?

```js
           postPolicyCategory(values)
                .then((response) => {
                    setSelectedCategory(response); // here instead of in finally block? reset the form values too?
                    setTimeout(refreshPolicyCategories, 200);
                    addToast('Successfully added category');
                })
```

**Review by pedrottimark — APPROVED** [review]
Keep up the steady steps forward. 2 questions which might affect future work.

File: ui/apps/platform/src/Containers/PolicyCategories/CreatePolicyCategoryModal.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                    resetForm();
+                    onClose();
+                });
+        },
+        validateOnMount: true,
+        validationSchema: yup.object().shape({
+            name: yup
+                .string()
+                .min(5, 'Policy category name must be at least 5 characters long')
+                .max(128, 'Policy category name must be less than 128 characters long')
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/PolicyCategories/CreatePolicyCategoryModal.tsx:63]: Is this max-length a backend requirement?
(128 seems like a random choice. I know relationship SQL, like Postgres, used to be optimized for a short string of up to 255 characters--haven't seen 128 before.)

**Review by vjwilson — APPROVED** [review]

File: ui/apps/platform/src/Containers/PolicyCategories/CreatePolicyCategoryModal.tsx (hunk 2 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/PolicyCategories/CreatePolicyCategoryModal.tsx:63]: Yes, backend requirement:

https://github.com/stackrox/stackrox/blob/master/central/policycategory/service/service_impl.go#L40

File: ui/apps/platform/src/Containers/PolicyCategories/PolicyCategoriesList.tsx (hunk 1 above)
Comment (alwayshooin) [reply ui/apps/platform/src/Containers/PolicyCategories/PolicyCategoriesList.tsx:29]: oh no it's residual from styling i thought i had to do. good eyes, i'll remove it

**Pull request merged**